### PR TITLE
(SIMP-6339) Fix Sensitive value on root password

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Fixed a bug where the root password field was attempting to set an 'undef'
  value as Sensitive.
- Bumped the supported Puppet version to include Puppet 6
- Removed Puppet 4 from the supported list

SIMP-6339 #close

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/54653644

File: functions/yum/repo/sanitize_simp_release_slug.pp [hunk 1]
```diff
@@ -9,6 +9,8 @@ function simp::yum::repo::sanitize_simp_release_slug(
   Variant[String,Undef] $simp_release_slug = undef
 ) {
 
+  inspect(simplib::simp_version())
```
Comment (jeannegreulich) [inline functions/yum/repo/sanitize_simp_release_slug.pp]: Is there a reason this is not named spaced?  Should it be simplib::inspect?  I could not find a puppet core function called inspect.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -241,43 +217,6 @@ pup5.5.7-remote_access_scenario:
...
-
-pup4.10-remote_access_scenario:
-  <<: *pup_4_10
-  <<: *acceptance_base
-  <<: *only_with_SIMP_FULL_MATRIX
-  script:
-    - 'bundle exec rake spec_clean'
-    - 'bundle exec rake beaker:suites[scenario_remote_access]'
-
 pup5.5.7-oel:
```
Comment (lnemsick-simp) [inline .gitlab-ci.yml]: Please test with Puppet 5.5.10 and Ruby 2.4.5, instead

File: functions/yum/repo/sanitize_simp_release_slug.pp (hunk 1 above)
Comment (trevor-vaughan) [reply functions/yum/repo/sanitize_simp_release_slug.pp]: This was a mistake. Good catch!

**Comment by trevor-vaughan** [comment]
> The latest release is 4.7 and the current version is 4.8. If we are not going to support puppet 4 anymore with this module should that be a major version bump to this module? It is a breaking change (if you are using puppet 4.)

It's covered under the version number allowances and is not a breaking change the way Semantic Versioning is written but only because we bring all updates needed and that the dropped software is not supported by anyone.

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/55786946

**Pull request merged**